This adds a call to notify_die() in the "no context" portion of
do_page_fault() as someone on the chain might care and want to do a fixup.

Signed-off-by: Tom Rini <trini@kernel.crashing.org>

 arch/x86_64/mm/fault.c      |    4 ++++
 include/asm-x86_64/kdebug.h |    1 +
 2 files changed, 5 insertions(+)

Index: linux-2.6.17/arch/x86_64/mm/fault.c
===================================================================
--- linux-2.6.17.orig/arch/x86_64/mm/fault.c	2006-07-06 13:12:40.000000000 +0530
+++ linux-2.6.17/arch/x86_64/mm/fault.c	2006-07-06 15:37:37.559660592 +0530
@@ -521,6 +521,10 @@
 	if (is_errata93(regs, address))
 		return; 
 
+	if (notify_die(DIE_PAGE_FAULT_NO_CONTEXT, "no context", regs,
+				error_code, 14, SIGSEGV) == NOTIFY_STOP)
+		return;
+
 /*
  * Oops. The kernel tried to access some bad page. We'll have to
  * terminate things with extreme prejudice.
Index: linux-2.6.17/include/asm-x86_64/kdebug.h
===================================================================
--- linux-2.6.17.orig/include/asm-x86_64/kdebug.h	2006-07-06 13:13:02.000000000 +0530
+++ linux-2.6.17/include/asm-x86_64/kdebug.h	2006-07-06 15:38:44.157536176 +0530
@@ -32,6 +32,7 @@
 	DIE_CALL,
 	DIE_NMI_IPI,
 	DIE_PAGE_FAULT,
+	DIE_PAGE_FAULT_NO_CONTEXT,
 };
 
 static inline int notify_die(enum die_val val, const char *str,
